balancer/endpointsharding/endpointsharding.go

```diff
@@ -258,8 +258,8 @@ func (p *pickerWithChildStates) Pick(info balancer.PickInfo) (balancer.PickResul
 	return picker.Pick(info)
 }
 
-// ChildStatesFromPicker returns the child states from the picker if the picker
-// is one provided in a call to UpdateState from this balancer.
+// ChildStatesFromPicker returns the child states from the picker that are
+// managed by this balancer.
```

Comment:
This reads weirdly now.  "returns the child states from the picker that are ___" -- to me "that are" applies to "the picker" and so doesn't disagree.

Maybe "returns the state of all the children managed by the endpoint sharding balancer that created this picker" or something like that?

Reply:
Switched to your suggestion.